ethcontract-common/src/artifact/hardhat.rs

```diff
@@ -139,42 +156,158 @@ impl HardHatLoader {
+    ///
+    /// [`allowed contracts`]: #structfield.contracts_allow_list
+    pub fn allow_contract(mut self, contract: impl Into<String>) -> Self {
+        self.contracts_allow_list.push(contract.into());
+        self
+    }
+
+    /// Add contract name to the list of [`denied contracts`].
+    ///
+    /// [`denied contracts`]: #structfield.contracts_deny_list
+    pub fn deny_contract(mut self, contract: impl Into<String>) -> Self {
+        self.contracts_deny_list.push(contract.into());
+        self
     }
 
-    /// Loads an artifact from bytes of JSON text.
-    pub fn load_from_slice(&self, v: &[u8]) -> Result<Artifact, ArtifactError> {
-        self.load_artifact("<unknown>", v, from_slice, from_slice)
+    /// Load an artifact from a JSON value.
+    pub fn load_from_reader(&self, f: Format, v: impl Read) -> Result<Artifact, ArtifactError> {
+        self.load_artifact(f, "<unknown>", v, from_reader, from_reader)
     }
 
-    /// Loads an artifact from string of JSON text.
-    pub fn load_from_str(&self, v: &str) -> Result<Artifact, ArtifactError> {
-        self.load_artifact("<unknown>", v, from_str, from_str)
+    /// Load an artifact from bytes of JSON text.
+    pub fn load_from_slice(&self, f: Format, v: &[u8]) -> Result<Artifact, ArtifactError> {
+        self.load_artifact(f, "<unknown>", v, from_slice, from_slice)
     }
 
-    /// Loads an artifact from a loaded JSON value.
-    pub fn load_from_value(&self, v: Value) -> Result<Artifact, ArtifactError> {
-        self.load_artifact("<unknown>", v, from_value, from_value)
+    /// Load an artifact from string of JSON text.
+    pub fn load_from_str(&self, f: Format, v: &str) -> Result<Artifact, ArtifactError> {
+        self.load_artifact(f, "<unknown>", v, from_str, from_str)
     }
 
-    /// Loads an artifact from disk.
-    pub fn load_from_file(&self, p: impl AsRef<Path>) -> Result<Artifact, ArtifactError> {
+    /// Load an artifact from a loaded JSON value.
+    pub fn load_from_value(&self, f: Format, v: Value) -> Result<Artifact, ArtifactError> {
+        self.load_artifact(f, "<unknown>", v, from_value, from_value)
+    }
+
+    /// Load an artifact from disk.
+    pub fn load_from_file(
+        &self,
+        f: Format,
+        p: impl AsRef<Path>,
+    ) -> Result<Artifact, ArtifactError> {
         let path = p.as_ref();
         let file = File::open(path)?;
         let reader = BufReader::new(file);
-        self.load_artifact(path.display(), reader, from_reader, from_reader)
+        self.load_artifact(f, path.display(), reader, from_reader, from_reader)
+    }
+
+    /// Load an artifact from `deployments` directory.
+    pub fn load_from_directory(&self, p: impl AsRef<Path>) -> Result<Artifact, ArtifactError> {
+        self._load_from_directory(p.as_ref())
+    }
+
+    fn _load_from_directory(&self, p: &Path) -> Result<Artifact, ArtifactError> {
+        let mut artifact = Artifact::with_origin(p.display().to_string());
```

Comment:
I'm super familiar with the hardhat directory layout, so this code was super easy to follow and understand. Maybe its worth including a comment blurb containing the "directory shape" for those who aren't so familiar. Alternatively, we could have a manually created test hardhat deployment directory that can be used with the unit test as well as document the expected "shape".